### DISCUSSION TITLE
[licence-gather-plugin] Upon failure to find license, indicate the dependency

### DISCUSSION THREAD

**Discussion by bric3** [vlsi/vlsi-release-plugins#56] [Q&A]
When running the license gather plugin, it may happen that `LICENSE-like` files are missing, but
the plugin does ot report which one, which is really tedious to identify if there are a lot of them.

```
> Task :modules:core:gatherLicense FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':modules:core:gatherLicense'.
> LICENSE-like files are missing
  ==============================
  
  BSD-3-Clause
  * org.pushing-pixels:radiance-animation:6.0.1
...
```

**Comment by vlsi** [comment]
`org.pushing-pixels:radiance-animation:6.0.1` does look like a dependency to me.

Could you please suggest which message would be better in that case?


The failure you have is that `radiance-animation` does list its license as BSD-3-Clause (I bet it is specified in pom.xml), however, the license text is not included into `radiance-animation.jar`.
If you shade (or include such `radiance-animation.jar` into your application) it would violate the terms of `BSD-3-Clause` because `BSD-3-Clause` requires including  **copyright notice** which is a part of the license.

**Comment by bric3** [comment]
Ahhh OK, I read this output wrong, I thought this dependency was correctly resolved. And didn't look at the jar.
Thank you very much for your clarification. Maybe the message might be improved to hint like you just did here.

Since I cannot fix the third party, maybe this can be fixed with something like (pseudo code as I'm on mobile now) : 

```
ignoreMissingLicenseFor.add(SpdxLicense.BSD_3_Clause.asExpression())
```

**Comment by vlsi** [comment]
> Since I cannot fix the third party

Well, you can put the license file under your source control under `licenses` folder: https://github.com/apache/jmeter/tree/master/src/licenses/licenses/hamcrest

By default, it would check `licenses/$artifactId` folder, and you can override it via `licenseFiles`

```kotlin
        overrideLicense("org.hamcrest:$lib:2.2") {
            // https://github.com/hamcrest/JavaHamcrest/issues/264
            // pom.xml lists "New BSD License", however it is BSD_3
            expectedLicense = SpdxLicense.BSD_3_Clause
            licenseFiles = "hamcrest"
        }
```

**Comment by bric3** [comment]
OK, great thanks a lot !

**Comment by bric3** [comment]
I guess the ticket can be closed ! 

I wonder if GH issues can be moved to discussions ?

**Reply by vlsi** [reply]
I've never tried that, so I clicked "migrate" to see what happens.
